### PR TITLE
Added consumer_group_metadata serdes to avoid bindings to hang on to object

### PR DESCRIPTION
**Pull request by edenhill**

### CODE REVIEW

File: src/rdkafka_cgrp.c [hunk 1]
```diff
@@ -3657,3 +3661,137 @@ rd_kafka_consumer_group_metadata_dup (
 }
 
 
+/*
+ * Consumer group metadata serialization format v1:
+ *  "CGMDv1:"<group_id>"\0"
+ * Where <group_id> is the group_id string.
+ */
+static const char rd_kafka_consumer_group_metadata_magic[7] = "CGMDv1:";
```
Comment (mhowlett) [inline src/rdkafka_cgrp.c]: not leaving anything to chance

**Comment by mhowlett** [comment]
LGTM. the magic thing seems a bit too paranoid, but i guess that fits with the general theme of the PR.

thanks!

**Comment by edenhill** [comment]
Paranoid? We need to identify the input as valid somehow.

**Comment by mhowlett** [comment]
> Paranoid? We need to identify the input as valid somehow.

it's externally visible, but effectively an internal detail, is where i was coming from

**Comment by edenhill** [comment]
The magic is defined in an internal source file, not in the public API.

**Pull request merged**